### PR TITLE
Fix: [Actions] nmlc version could not be determined during regression

### PR DESCRIPTION
**Pull request by glx22**
Regression log is very noisy with each test triggering
```
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
Git checkout found but cannot determine its version. Error:  Command '['git', '-C', '/Users/runner/work/nml/nml', 'describe', '--tags', '--long']' returned non-zero exit status 128.
Git checkout found but cannot determine its version. Error:  Command '['git', '-C', '/Users/runner/work/nml/nml', 'describe', '--tags', '--long']' returned non-zero exit status 128.
```

Fetch more data during checkout so `git describe` can do its job.

### CODE REVIEW

File: .github/workflows/regression.yml [hunk 1]
```diff
@@ -26,6 +26,13 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
```
Comment (LordAro) [inline .github/workflows/regression.yml:30]: Why is this required?

Comment (glx22) [reply .github/workflows/regression.yml:30]: Because with default depth (1), you only get the merge commit, and its relation with last tag is impossible to find

**Review by rubidium42 — APPROVED** [review]
